### PR TITLE
Integrate fireblocks wallet into ejector

### PR DESCRIPTION
**Pull request by ian-shim**
## Why are these changes needed?
Integrate Fireblocks wallet into ejector

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: disperser/dataapi/ejector.go [hunk 1]
```diff
@@ -86,11 +106,73 @@ func (e *ejector) eject(ctx context.Context, nonsigningRate *OperatorsNonsigning
...
+		txID, err = e.wallet.SendTransaction(ctx, txn)
+		var urlErr *url.Error
+		didTimeout := false
+		if errors.As(err, &urlErr) {
+			didTimeout = urlErr.Timeout()
+		}
+		if didTimeout || errors.Is(err, context.DeadlineExceeded) {
+			e.logger.Warn("failed to send txn due to timeout", "hash", txn.Hash().Hex(), "numRetries", retryFromFailure, "maxRetry", maxSendTransactionRetry, "err", err)
+			retryFromFailure++
+			continue
```
Comment (jianoaix) [inline disperser/dataapi/ejector.go:136]: No need "continue"

File: disperser/dataapi/ejector.go [hunk 2]
```diff
@@ -86,11 +106,73 @@ func (e *ejector) eject(ctx context.Context, nonsigningRate *OperatorsNonsigning
...
+		if err != nil {
+			return fmt.Errorf("failed to update gas price: %w", err)
+		}
+		txID, err = e.wallet.SendTransaction(ctx, txn)
+		var urlErr *url.Error
+		didTimeout := false
+		if errors.As(err, &urlErr) {
+			didTimeout = urlErr.Timeout()
+		}
+		if didTimeout || errors.Is(err, context.DeadlineExceeded) {
```
Comment (jianoaix) [inline disperser/dataapi/ejector.go:133]: So one timeout is from connecting to fireblocks endpoint, the other the fireblocks itself couldn't send txn internally?

File: disperser/cmd/dataapi/main.go [hunk 3]
```diff
@@ -147,3 +160,70 @@ func RunDataApi(ctx *cli.Context) error {
...
+		sender, err := wallet.SenderAddress(context.Background())
+		if err != nil {
+			return nil, err
+		}
+		if sender.Cmp(gethcommon.HexToAddress(config.FireblocksConfig.WalletAddress)) != 0 {
+			return nil, fmt.Errorf("configured wallet address %s does not match derived address %s", config.FireblocksConfig.WalletAddress, sender.Hex())
+		}
+		logger.Info("Initialized Fireblocks wallet", "vaultAccountName", config.FireblocksConfig.VaultAccountName, "address", sender.Hex())
+	} else if len(config.EthClientConfig.PrivateKeyString) > 0 {
+		privateKey, err := crypto.HexToECDSA(config.EthClientConfig.PrivateKeyString)
```
Comment (jianoaix) [inline disperser/cmd/dataapi/main.go:207]: wonder if there is any benefit if we put raw private key into the secret manager like the api keys above?

**Review by jianoaix — COMMENTED** [review]
Lg, thanks for adding this!

File: disperser/dataapi/ejector.go (hunk 2 above)
Comment (ian-shim) [reply disperser/dataapi/ejector.go:133]: `didTimeout` detects whether it timed out from the http client, which can happen with fireblocks API call. `context.DeadlineExceeded` detects whether it timed out from from rpc calls, which can happen if wallet is using ethClient to send transactions

File: disperser/cmd/dataapi/main.go (hunk 3 above)
Comment (ian-shim) [reply disperser/cmd/dataapi/main.go:207]: Yes. That would be arguably more secure, but it adds a dependency on AWS secret manager so neither wallet will be functional if secret manager is down

**Review by jianoaix — APPROVED** [review]

**Pull request merged**